// numabbr/es/index.js
import { round, commatize } from './utils';
export class NumberAbbreviate {
    constructor(units = {}) {
        this.units = Object.keys(units)
            .map((unit) => ({ unit, value: units[unit] }))
            .sort((a, b) => a.value - b.value);
    }
    abbreviate(num, options = {}) {
        const { precision = 2, commatize: commaOptions } = options;
        const negative = num < 0;
        const raw = Math.abs(num);
        for (let i = this.units.length - 1; i >= 0; i -= 1) {
            const unit = this.units[i].unit;
            const size = this.units[i].value;
            if (raw >= size) {
                const result = round(raw / size, precision);
                return ((negative ? '-' : '') +
                    (commaOptions
                        ? commatize(result, commaOptions)
                        : result) +
                    unit);
            }
        }
        return `${num}`;
    }
}
const numabbr = new NumberAbbreviate({
    K: 1000,
    M: 1000000,
    B: 1000000000,
    T: 1000000000000,
});
export default function (num, options) {
    return numabbr.abbreviate(num, options);
}
